### PR TITLE
receiver/prometheus: avoid nil pointer deref when adjusting metric points

### PR DESCRIPTION
**Pull request by Aneurysm9**
**Description:** Fixes bounds check and nil pointer dereferences that may occur when adjusting histogram and summary metric points.

**Link to tracking Issue:** #3058

### CODE REVIEW

**Review by rakyll — APPROVED** [review]

File: receiver/prometheusreceiver/internal/metrics_adjuster.go [hunk 1]
```diff
@@ -286,14 +286,26 @@ func (ma *MetricsAdjuster) isReset(metricType metricspb.MetricDescriptor_Type,
...
+		if (currentSummary.Count != nil &&
+			previousSummary.Count != nil &&
+			currentSummary.Count.GetValue() < previousSummary.Count.GetValue()) ||
+
+			(currentSummary.Sum != nil &&
+				previousSummary.Sum != nil &&
+				currentSummary.Sum.GetValue() < previousSummary.Sum.GetValue()) {
 			// reset detected
 			return false
 		}
```
Comment (odeke-em) [inline receiver/prometheusreceiver/internal/metrics_adjuster.go:311]: We can make this much cleaner by a reverse conditional
```go
if currentSummary.Count == nil || previousSummary.Count == nil {
   return true
}
if currentSummary.Count.GetValue() < previousSummary.Count.GetValue() {
   return false
}
if currentSummary.Sum == nil || previousSummary.Sum == nil {
   return true
}
return currentSummary.Sum.GetValue() >= previousSummary.Sum.GetValue()
```

**Review by odeke-em — APPROVED** [review]
Thank you for this work @Aneurysm9! LGTM but with suggestions to refactor to make early returns and reducing nesting/conditions.

File: receiver/prometheusreceiver/internal/metrics_adjuster.go (hunk 1 above)
Comment (Aneurysm9) [reply receiver/prometheusreceiver/internal/metrics_adjuster.go:311]: I'm not sure that is functionally equivalent.  It seems that the count and sum attributes on summary datapoints are both optional.  We should still detect resets when only one of them is present.

Comment (odeke-em) [reply receiver/prometheusreceiver/internal/metrics_adjuster.go:311]: Please tailor it to the equivalent, but the point still stands we can avoid that overly involved single line comparison.

**Comment by odeke-em** [comment]
@bogdandrutu @tigrannajaryan @alolita @anuraaga please take a look, this change is ready for merge and was approved by some of us. Thank you!

**Review by tigrannajaryan — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
Loadtest failed and for some reasons I can't restart CircleCI. Please touch the commit to trigger the CI.

**Comment by odeke-em** [comment]
@Aneurysm9 please force push a commit to address @tigrannajaryan's request.

**Pull request merged**